### DISCUSSION TITLE
Accessing images from README.md of Private Repos

### DISCUSSION THREAD

**Discussion by inanperillo** [community/community#23385] [New to GitHub]
<p>My team has a private GitHub repository and has an internal wiki via GraphQL.<br>
We have a python script that pulls README.md of specified repos that are to be placed in wiki via PAT.<br>
We are able download the text contents of the README.md and see it in the wiki.<br>
The problem is the images in these README.md files does not show.<br>
The images in these README.md files use the format:<br>
!(My Image)[<a href="https://github.com/%7Benterprise-private%7D/%7Bmy-repo%7D/blob/main/static/img/%7Bimage.jpg%7D" rel="noopener nofollow ugc">https://github.com/{enterprise-private}/{my-repo}/blob/main/static/img/{image.jpg}</a>]</p>
<p>When accessing the images while not logged in, it does indeed show 404 Error.<br>
<a href="https://github.com/%7Benterprise-private%7D/%7Bmy-repo%7D/blob/main/static/img/%7Bimage.jpg%7D" class="onebox" target="_blank" rel="noopener nofollow ugc">https://github.com/{enterprise-private}/{my-repo}/blob/main/static/img/{image.jpg}</a></p>
<p>Is  it possible to access the image without having the need to login? Or this is strictly forbidden since it’s a private repository?</p>

**Comment by tajmone** [comment, marked as answer]
<aside class="quote no-group" data-username="inanperillo" data-post="1" data-topic="189336">
<div class="title">
<div class="quote-controls"></div>
 inanperillo:</div>
<blockquote>
<p>Is it possible to access the image without having the need to login? Or this is strictly forbidden since it’s a private repository?</p>
</blockquote>
</aside>
<p>The latter, since all contents are private. If you want to make part of the docs and images visible publicly you’ll have to come up with an automated solution to export them somewhere else, e.g. a public repository, a website, etc.</p>
<p>You can easily achieve this via some script, which you can then exploit via some CI service or GitHub Actions to ensure that whenever the master branch of you repository is updated all the public assets are update too (you can use Git commands in the script to detect changed files).</p>

**Comment by inanperillo** [comment]
<p>Thanks for the reply. This is what I thought of too. We don’t need all repos in enterprise-private, only a selected few anyways. So I know this can be done.</p>
<p>I guess, what I’m asking here, and I can’t find info about it, is:</p>
<p>Can i pass PAT in the URL as an html content for the readme?</p>
<p>I was able to do curl commands using <a href="mailto:PAT@raw.githubusercontent.com">PAT@raw.githubusercontent.com</a> in CLI but in my testing have failed to use this as an html content (as part of the readme).</p>
<p>Not that I need this in the solution, but now, more of: I’m interested to know.</p>

**Comment by tajmone** [comment]
<aside class="quote no-group" data-username="inanperillo" data-post="3" data-topic="189336">
<div class="title">
<div class="quote-controls"></div>
 inanperillo:</div>
<blockquote>
<p>Can i pass PAT in the URL as an html content for the readme?</p>
</blockquote>
</aside>
<p>I’m not sure of this, for it might require some redirection steps after validation, which are only possible to handle if you have full control over a server. Maybe this page is of help, or at least provides a good introduction and useful external links:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://plaid.com/docs/link/oauth/">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/11064292/181076691-f2460730-a9b1-48fa-8491-95d227f56a95.png" class="site-icon" width="32" height="32">

      <a href="https://plaid.com/docs/link/oauth/" target="_blank" rel="noopener nofollow ugc">plaid.com</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/360;"><img src="https://user-images.githubusercontent.com/11064292/181076692-6c813a2b-08ea-4a2e-b8f2-31ba9a70f5e1.jpeg" class="thumbnail" width="690" height="360" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/9/c/9c1f9bc5467d08d5e7d0df22e8b2c1f2fb22d8b2_2_690x360.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/9/c/9c1f9bc5467d08d5e7d0df22e8b2c1f2fb22d8b2_2_1035x540.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/9/c/9c1f9bc5467d08d5e7d0df22e8b2c1f2fb22d8b2_2_1380x720.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/c/9c1f9bc5467d08d5e7d0df22e8b2c1f2fb22d8b2_2_10x10.png"></div>

<h3><a href="https://plaid.com/docs/link/oauth/" target="_blank" rel="noopener nofollow ugc">Link</a></h3>

  <p>Configure Link to connect to institutions via OAuth, which redirects users to their bank's site to log in. OAuth is required for European institutions.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Whether it’s doable or not, I don’t think that including your username and authentication token inside a link within a public document is a good idea, for it would compromise the security of that account. Also, this might violate GitHub EULA, since you’re expected to do your best to protect all your secret keys and access tokens.</p>

**Comment by inanperillo** [comment]
<p>That makes sense.</p>
<p>With the absence of documentation about it and with the points you raised, I’d assume this is indeed not possible.</p>
<p>Thanks!</p>

**Comment by EssamWisam** [comment]


**Reply by airtower-luna** [reply]
If you open the image from a private repository in the browser, the URL includes an automatic token (not a PAT) that allows access *temporarily*. I'm not sure exactly how long it's valid, but it should fail after a little while. :wink:

**Comment by dotku** [comment]
Okay, what I have done here is create a new repo that would be used for sharing public assets
